File: revision/practise_exam/q33.pl
#!/usr/local/bin/perl -w

$filename = $ARGV[0];
$to_match = $ARGV[1];

open F, "<", $filename or die "Cannot open file.";
foreach $line (<F>) {
    chomp $line;
    while ($line =~ /(.*?[^(]*)($to_match)([^)]*.*)/) {
        ($bef, $match, $aft) = ($1, $2, $3);
        print "bef = $bef, match = $match, aft = $aft\n";
        $line = "$bef($match)$aft";
    }
    print "line = $line\n";
}

close F;

File: revision/practise_exam/q33.py
#!/usr/bin/python3.6

import sys, re
to_match = sys.argv[2]
file = open(sys.argv[1])

for line in file.readlines():
    line = line.strip()
    match = re.search("({0})".format(to_match), line)
    if match:
        match = match.group(1)
        line = re.sub(match, "(" + match + ")", line)
        print (line)
    

